crates/oxpinyin-store/src/tkrzw/shim.cc

```
@@ -0,0 +1,258 @@
+  }
+}
+
+// Maps an exception escaping an entry point onto the status channel:
+// UNKNOWN_ERROR ("generic error whose cause is unknown") fits an
+// allocation failure or similar. Callers report any non-SUCCESS code
+// verbatim, so nothing here has to be named on the Rust side. Building
+// the message can itself throw under memory exhaustion; that terminates
+// the process, which is still better than unwinding into Rust.
+ShimStatus wrap(const tkrzw::Status& status);
+
+ShimStatus caught(const std::exception& e) {
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)